src/System.Management.Automation/engine/regex.cs

```diff
@@ -237,29 +232,15 @@ internal static string Escape(string pattern, char[] charsNotToEscape)
             {
                 char ch = pattern[i];
 
-                //
-                // if it is a wildcard char, escape it
-                //
-                if (IsWildcardChar(ch) && !charsNotToEscape.Contains(ch))
+                if (charsToEscape.Contains(ch))
```

Comment:
The change actually makes this method run slower for the common usage of it. All the call sites of this method other than 1 uses an empty array for `charsNotToEscape`, because the check of `IsWildcardChar` should be faster than calling `Contains` on a span.

Reply:
Above there is fast return for common case (no wildchars):
```c#
            if (pattern == string.Empty || charsToEscape.IsEmpty || pattern.AsSpan().IndexOfAny(charsToEscape) < 0)
            {
                return pattern;
            }
```
Contains() method is very optimized in .Net. SSE works for "*?[]". Perhaps it is worse for "[]".
But in general it does not matter because it is used in path processing, which is so slow that a loss of 20 ns makes no difference.

Reply:
`pattern.AsSpan().IndexOfAny(charsToEscape)` will do a traverse on the characters, so how could it be faster than doing a traverse directly in the code? Also, in case of the path contains wildcards, you will be doing 2 travers on the characters, once from `IndexOfAny`, and one from the `for` loop below.

It's not about whether or not the slowness is noticeable, but we should not change to something slower.

Reply:
I assume that these symbols are very rarely used in paths in practice. This means that this method works faster than the previous one.

> It's not about whether or not the slowness is noticeable, but we should not change to something slower.

This cannot be an argument after MSFT team has added case normalization for all paths and kill script performance at all.